### PR TITLE
device_model: replace pngs with inline ditaa diagrams

### LINKED ISSUE #191: Convert labs png diagrams to ditaa
There are multiple png images in the labs containing diagrams that can be converted to ditaa.
As suggested by Tavi: "We can use http://asciiflow.com/ to start the skeleton for the ascii diagrams."

### PR DESCRIPTION
**Pull request by tavip**
Fixes #191

### CODE REVIEW

File: Documentation/teaching/labs/device_model.rst [hunk 1]
```diff
@@ -90,7 +90,37 @@ are as follows:
...
+              +--+--+                                             +---+---+                            +----+----+                                +---+----+
+                 |                                                    |                                     |                                         |
+                 |                                                    |                                     |                       +-----------------+-----------------+
+                 |                                                    |                                     |                       |                                   |
+                 v                                                    v                                     v                       v                                   v
+   +--------------------------------+                      +-----------------------+            +-----------------------+     +----------------------+    +-------------------------+
+   | mydriver: struct device_driver |                      | myclass: struct class |            | mybus0: struct device |     | mybus: struct module |    | mydriver: struct module |
```
Comment (valighita) [inline Documentation/teaching/labs/device_model.rst:109]: In the old image, the first item was `mybus: struct bus_type`

File: Documentation/teaching/labs/device_model.rst [hunk 2]
```diff
@@ -90,7 +90,37 @@ are as follows:
...
+.. ditaa::
+                                                                   +------+
+                                                                   | /sys |
+                                                                   +--+---+
+                                                                      |
+                 +----------------------------------------------------+-------------------------------------+-----------------------------------------+
+                 |                                                    |                                     |                                         |
+                 v                                                    v                                     v                                         v
+              +-----+                                             +-------+                            +---------+                                +--------+ 
+              | bus |                                             | class |                            | de|ices |                                | module |
```
Comment (valighita) [inline Documentation/teaching/labs/device_model.rst:102]: Typo: `devices`

File: Documentation/teaching/labs/device_model.rst (hunk 1 above)
Comment (tavip) [reply Documentation/teaching/labs/device_model.rst]: fixed

File: Documentation/teaching/labs/device_model.rst (hunk 2 above)
Comment (tavip) [reply Documentation/teaching/labs/device_model.rst]: fixed

**Pull request merged**